### PR TITLE
ref(ts) Convert platform data to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/data/platforms.tsx [hunk 1]
```diff
@@ -20,4 +21,4 @@ export default [].concat(
   ...[...platforms, otherPlatform].map(platform =>
     platform.integrations.map(i => ({...i, language: platform.id}))
   )
-);
+) as PlatformIntegration[];
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/data/platforms.tsx:24]: heh. I think I tried to do this one even fancier and had a hard time getting this concat to be happy. This is probably fine though

**Review by evanpurkhiser — APPROVED** [review]

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/data/platforms.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/data/platforms.tsx:24]: I thought of typing some of the intermediate steps but it didn't really change the end result, and I figured this was better than any.

**Comment by markstory** [comment]
I had to make a few changes to the tsconfig.json in order to get builds and typechecking to pass.

**Review by billyvg — APPROVED** [review]

**Comment by markstory** [comment]
Both the vercel and cloud builder errors are related to this change :cry:

**Comment by billyvg** [comment]
@markstory weird, how come the github action runs it successfully?

**Comment by markstory** [comment]
@billyvg In the gh action, the tests directory is available. In both vercel and docker only the src directory is mounted and tsc can't find the files. I can see if having both the real and test paths helps.

**Comment by billyvg** [comment]
Can we perhaps `declare module`?

**Comment by markstory** [comment]
@billyvg I don't think that is necessary, I got tsc to be happy by setting both the fixtures and real file as paths for the module alias.

**Pull request merged**